### PR TITLE
Solr rackspace template

### PR DESCRIPTION
**Pull request by amontenegro**

### CODE REVIEW

File: puppet/modules/orcid_openjdk/manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    creates	=> "/etc/apt/sources.list.d/openjdk-r-ppa-trusty.list",
+    require	=> File["/usr/lib/jvm"]
+  }
+  
+  exec { "apt-get-update":
+    command	=> "apt-get update",
+    require	=> Exec["add-openjdk-ppa-repo"]
+  }
+  
+  exec { "install-openjdk-11":
```
Comment (jpeerz) [inline puppet/modules/orcid_openjdk/manifests/init.pp]: use package { } instead

File: puppet/modules/orcid_solr/manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  file { "${solr_home}/git":
+    ensure	=> directory,
+    owner	=> $solr_user,
+    group	=> $solr_user,
+    mode	=> "0755",
+    require	=> File[$solr_home]
+  }
+  
+  vcsrepo { "${solr_home}/git/ORCID-Source":
```
Comment (jpeerz) [inline puppet/modules/orcid_solr/manifests/init.pp]: please fix tabs/spaces

File: puppet/modules/orcid_openjdk/templates/scripts/install_letsencrypt_cert.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+wget https://letsencrypt.org/certs/isrgrootx1.pem;
```
Comment (jpeerz) [inline puppet/modules/orcid_openjdk/templates/scripts/install_letsencrypt_cert.erb]: we could also use file { for all wget calls }

File: puppet6/environments/dev/manifests/site.pp [hunk 4]
```diff
@@ -46,5 +46,20 @@
...
 
+    include orcid_solr
 }
+
+node 'solr-2.orcid.org' {
+
+    $subdomain_name = 'solr2test'
+    $server_group   = 'solr-slave'
+
+	include orcid_solr
```
Comment (jpeerz) [inline puppet6/environments/dev/manifests/site.pp]: fix spacing

File: puppet6/environments/dev/modules/orcid_openjdk/manifests/init.pp [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class orcid_openjdk {
+  
+  include openjdk::package
```
Comment (jpeerz) [inline puppet6/environments/dev/modules/orcid_openjdk/manifests/init.pp]: at the end is java, I suggest move those manifest into java | orcid_java modules, so we call
`include orcid_java::openjdk` or just `include orcid_java`

File: puppet6/environments/dev/modules/orcid_solr/manifests/config.pp [hunk 6]
```diff
@@ -0,0 +1,95 @@
+class orcid_solr::config (
+    $solr_home		= "/opt/solr",
+    $solr_user		= "orcid_tomcat",
+    $solr_version	= "8.0.0",
+    $solr_port		= "8983",
+	$solr_tar_file	= "/tmp/solr-${solr_version}.tgz"
```
Comment (jpeerz) [inline puppet6/environments/dev/modules/orcid_solr/manifests/config.pp]: fix spacing

File: puppet6/environments/dev/modules/orcid_solr/manifests/config.pp [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+  file { "${solr_home}/git":
+    ensure	=> directory,
+    owner	=> $solr_user,
+    group	=> $solr_user,
+    mode	=> "0755",
+    require	=> File[$solr_home]
+  }
+  
+  vcsrepo { "${solr_home}/git/ORCID-Source":
```
Comment (jpeerz) [inline puppet6/environments/dev/modules/orcid_solr/manifests/config.pp]: fix spacing

File: puppet6/environments/dev/modules/orcid_solr/manifests/init.pp [hunk 8]
```diff
@@ -0,0 +1,9 @@
+class orcid_solr {
+
+  include orcid_users::tomcat
+  include orcid_baseapps
+  include orcid_openjdk
+  include orcid_python
+  include orcid_solr::config
```
Comment (jpeerz) [inline puppet6/environments/dev/modules/orcid_solr/manifests/init.pp]: orcid_solr should be including/extends `solr` and `orcid_solr::config`

File: puppet6/environments/production/modules/orcid_solr/manifests/config.pp [hunk 9]
```diff
@@ -0,0 +1,95 @@
+class orcid_solr::config (
+    $solr_home		= "/opt/solr",
+    $solr_user		= "orcid_tomcat",
+    $solr_version	= "8.0.0",
+    $solr_port		= "8983",
+	$solr_tar_file	= "/tmp/solr-${solr_version}.tgz"
```
Comment (jpeerz) [inline puppet6/environments/production/modules/orcid_solr/manifests/config.pp]: spacing

File: puppet6/environments/production/modules/orcid_solr/manifests/config.pp [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    group	=> $solr_user,
+    mode	=> "0755",
+    require	=> File[$solr_home]
+  }
+  
+  vcsrepo { "${solr_home}/git/ORCID-Source":
+    ensure		=> latest,
+    owner		=> $solr_user,
+    group		=> $solr_user,
+    provider	=> git,
```
Comment (jpeerz) [inline puppet6/environments/production/modules/orcid_solr/manifests/config.pp]: spacing

File: rack/orcidSolr.yaml [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+      cloud_config:
+        write_files:
+        - encoding: b64
+          path: /opt/init.sh
+          content: IyEvYmluL2Jhc2gKCiMgVXBkYXRlIGhvc3RuYW1lCkRPTUFJTj0ib3JjaWQub3JnIgpIT1NUTkFNRT1gL2Jpbi9ob3N0bmFtZWAKRlFETj0iJEhPU1ROQU1FLiRET01BSU4iCmNhdCA+IC9ldGMvaG9zdG5hbWUgPDxFT1QKJEZRRE4KRU9UCi9iaW4vaG9zdG5hbWUgLUYgL2V0Yy9ob3N0bmFtZQplY2hvICJbT0tdICQoL2Jpbi9ob3N0bmFtZSkiCgojIFVwZGF0ZSBwdXBwZXQgY2VydG5hbWUKbXYgL29wdC9wdXBwZXQudGVtcC5jb25mIC9ldGMvcHVwcGV0bGFicy9wdXBwZXQvcHVwcGV0LmNvbmYKc2VkIC1pICJzfE5PREVOQU1FfCRGUUROfGciIC9ldGMvcHVwcGV0bGFicy9wdXBwZXQvcHVwcGV0LmNvbmYKZWNobyAiW09LXSAkKGdyZXAgY2VydG5hbWUgL2V0Yy9wdXBwZXRsYWJzL3B1cHBldC9wdXBwZXQuY29uZikiCgojIHh2ZGIgbm9uLWludGVyYWN0aXZlIHBhcnRpdGlvbmluZwp3aGlsZSBbIGAvYmluL2xzYmxrIHwgZ3JlcCB4dmRiIHwgd2MgLWxgIC1lcSAwIF07ZG8gCiAgICBlY2hvICd3YWl0aW5nICcKICAgIHNsZWVwIDUKZG9uZQpvcmNpZF9kYXRhX2Rpc2s9L2Rldi94dmRiCnByaW50ZiAib1xublxucFxuMVxuXG5cbndcbiIgfCBmZGlzayAiJG9yY2lkX2RhdGFfZGlzayIKc2xlZXAgMwpta2ZzIC10IGV4dDMgIiR7b3JjaWRfZGF0YV9kaXNrfTEiCmVjaG8gIiR7b3JjaWRfZGF0YV9kaXNrfTEgL21udC9vcmNpZF9kYXRhIGV4dDMgZGVmYXVsdHMgIDAgMiIgPj4gL2V0Yy9mc3RhYgptb3VudCAtYQpkZiAtaAplY2hvICJbT0tdIHh2ZGIgc2V0dXAgJChkZiAtaCkiCmFwdC1nZXQgLXkgYXV0b3JlbW92ZQ==
```
Comment (jpeerz) [inline rack/orcidSolr.yaml]: Let's include the content decoded as a sh file in rack folder.

**Comment by amontenegro** [comment]
Hi @simplefxone,

Could you please take a look at this? 

Thanks

**Comment by amontenegro** [comment]
[Upgrade solr libraries (11)](https://trello.com/c/1fggAEVk/5391-upgrade-solr-libraries-11)

**Comment by amontenegro** [comment]
Great @simplefxone! could you please merge the PR?

Thanks

**Pull request merged**